webapp/src/test/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionServiceTest.java

```diff
@@ -1365,6 +1370,51 @@ public void testTextUnitsWithMultipleUsagesForPlural() throws Exception {
     checkAssetTextUnits(assetTextUnits, expectedUsages);
   }
 
+  @Test
+  public void testBranchesWithSameStrings() throws Exception {
+    final RepositoryStatisticsJobScheduler mockedScheduler =
+        Mockito.mock(RepositoryStatisticsJobScheduler.class);
+    this.assetExtractionService.setRepositoryStatisticsJobScheduler(mockedScheduler);
```

Comment:
Ah I think I see why you'd the set method earlier, if you Autowire the bean and don't set it as a private field it'll be protected access, so you can set it directly in the test as it's the same package using `this.assetExtractionService.repositoryStatisticsJobScheduler = mockedScheduler`

Reply:
I tried that, but the mock object was not being used by the AssetExtractionService object, and this assert `verify(mockedScheduler, times(1)).schedule(anyLong());` always failed

Reply:
**@MockBean** and **@SpyBean** caused several errors with the following message:
`org.springframework.jdbc.CannotGetJdbcConnectionException: Failed to obtain JDBC Connection`

Reply:
> I would still expect that you're able to Autowire the RepositoryStatisticsJobScheduler in the AssetExtractionService via field injection, drop the private access modifier against it and then you should be able to set the field directly in your test (not using a @MockBean, just using a standard mock as you do in the set method approach).
After making that change, I am getting this error message:

> ERROR] Failures: 
[ERROR] com.box.l10n.mojito.service.assetExtraction.AssetExtractionServiceTest.testBranchesWithSameStrings
[ERROR]   Run 1: AssetExtractionServiceTest.testBranchesWithSameStrings:1419 
Wanted but not invoked:
repositoryStatisticsJobScheduler.schedule(
    <any long>
);
-> at com.box.l10n.mojito.service.repository.statistics.RepositoryStatisticsJobScheduler.schedule(RepositoryStatisticsJobScheduler.java:21)
Actually, there were zero interactions with this mock.

[ERROR]   Run 2: AssetExtractionServiceTest.testBranchesWithSameStrings:1419 
Wanted but not invoked:
repositoryStatisticsJobScheduler.schedule(
    <any long>
);
-> at com.box.l10n.mojito.service.repository.statistics.RepositoryStatisticsJobScheduler.schedule(RepositoryStatisticsJobScheduler.java:21)
Actually, there were zero interactions with this mock.
For some reason, setting the field directly in the test did not work; the mock object is not being used during execution, but the real object is.